From 566017641edbafbd06df3443bf903827b35813cb Mon Sep 17 00:00:00 2001
From: dineshsv <dineshsv@ami.com>
Date: Wed, 21 Sep 2022 15:17:16 +0530
Subject: [PATCH] For Get-set-user payload acess,created users will  get
 response as expected for non-created users will get response as Invalid data
 field request.

Tested:

 -verified Get-Set-user-payload command is working as expected.

msgubuntu4@msgubuntu4-OptiPlex-5080:~$ ipmitool -H 10.0.123.234 -U root -P 0penBmc -I lanplus -C 17 raw 0x06 0x4d 0x03 0x01
 02 00 00 00
msgubuntu4@msgubuntu4-OptiPlex-5080:~$ ipmitool -H 10.0.123.234 -U root -P 0penBmc -I lanplus -C 17 raw 0x06 0x4d 0x03 0x02
Unable to send RAW command (channel=0x0 netfn=0x6 lun=0x0 cmd=0x4d rsp=0xcc): Invalid data field in request
msgubuntu4@msgubuntu4-OptiPlex-5080:~$ ipmitool -H 10.0.123.234 -U root -P 0penBmc -I lanplus -C 17 raw 0x06 0x4d 0x03 0x03
Unable to send RAW command (channel=0x0 netfn=0x6 lun=0x0 cmd=0x4d rsp=0xcc): Invalid data field in request
msgubuntu4@msgubuntu4-OptiPlex-5080:~$ ipmitool -H 10.0.123.234 -U root -P 0penBmc -I lanplus -C 17 raw 0x06 0x4d 0x03 0x04
Unable to send RAW command (channel=0x0 netfn=0x6 lun=0x0 cmd=0x4d rsp=0xcc): Invalid data field in request
msgubuntu4@msgubuntu4-OptiPlex-5080:~$ ipmitool -H 10.0.123.234 -U root -P 0penBmc -I lanplus -C 17 raw 0x06 0x4d 0x03 0x05
Unable to send RAW command (channel=0x0 netfn=0x6 lun=0x0 cmd=0x4d rsp=0xcc): Invalid data field in request

Signed-off-by: dineshsv <dineshsv@ami.com>
---
 user_channel/user_layer.cpp | 12 ++++++++++++
 1 file changed, 12 insertions(+)

diff --git a/user_channel/user_layer.cpp b/user_channel/user_layer.cpp
index bb9ac38..ce2d500 100644
--- a/user_channel/user_layer.cpp
+++ b/user_channel/user_layer.cpp
@@ -185,6 +185,12 @@ Cc ipmiUserSetUserPayloadAccess(const uint8_t chNum, const uint8_t operation,
                                 const uint8_t userId,
                                 const PayloadAccess& payloadAccess)
 {
+	std::string userName;
+   ipmiUserGetUserName(userId,userName);
+     if(userName.empty())
+     {
+       return ccInvalidFieldRequest;
+     }
 
     if (!UserAccess::isValidChannel(chNum))
     {
@@ -202,6 +208,12 @@ Cc ipmiUserSetUserPayloadAccess(const uint8_t chNum, const uint8_t operation,
 Cc ipmiUserGetUserPayloadAccess(const uint8_t chNum, const uint8_t userId,
                                 PayloadAccess& payloadAccess)
 {
+     std::string userName;
+   ipmiUserGetUserName(userId,userName);
+     if(userName.empty())
+     {
+       return ccInvalidFieldRequest;
+     }
 
     if (!UserAccess::isValidChannel(chNum))
     {
-- 
2.25.1

